### PR TITLE
Backport Finder and Config classes from 2.x line

### PR DESCRIPTION
**Pull request by keradus**
As on 2.x line we want to remove `Symfony\CS\Config\Config` and use only `Symfony\CS\Config` it would be great to allow to switch to it already on 1.x line, same for Finder.

// requested by @GrahamCampbell 

ref #1646

### CODE REVIEW

File: Symfony/CS/Config.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
+<?php
```
Comment (keradus) [inline Symfony/CS/Config.php]: This file was renamed from `Symfony/CS/Config/Config.php`, then:
- change namespace

File: Symfony/CS/Finder.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<?php
```
Comment (keradus) [inline Symfony/CS/Finder.php]: This file was renamed from `Symfony/CS/Finder/DefaultFinder.php`, then:
- change namespace
- stop searching for `*.xml` and `*.yml` as 2.x Finder does not do that, DefaultFinder will still search for them to not break BC

**Comment by keradus** [comment]
ping @GrahamCampbell  ;)

**Comment by keradus** [comment]
Any thoughts?

**Comment by SpacePossum** [comment]
like the idea :)
however.. on 1.x this will break for others with configs like https://github.com/GeckoPackages/GeckoTwig/blob/master/.php_cs.dist#L14 ?

**Comment by keradus** [comment]
Why is that?
The class is still there: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1670/files#diff-c3d879ff174f5aec5f64e9fdb6cfe4a9R22 , only deprecated

**Comment by SpacePossum** [comment]
indeed, nevermind me :)
lgtm :+1:

**Comment by SpacePossum** [comment]
Thanks :)

**Pull request closed without merging**